t/backup_expiration.t

```diff
@@ -119,74 +119,87 @@ describe "HTFeed::BackupExpiration" => sub {
-      is($deleted, 0, 'object is not deleted');
-      ok(!mets_deleted($storage), "new ($storage->{timestamp}) mets left intact");
-      ok(!zip_deleted($storage), "new ($storage->{timestamp}) zip left intact");
+      my $deleted = count_deleted_objects('test', 'test');
+      is($deleted, 0, 'no objects deleted');
+      foreach my $storage (@versions) {
+        ok(!mets_deleted($storage), "new ($storage->{timestamp}) mets left intact");
+        ok(!zip_deleted($storage), "new ($storage->{timestamp}) zip left intact");
+      }
     };
 
-    it "deletes old versions when there is a new one" => sub {
-      my @old_versions;
-      foreach my $n (1 .. 2) {
-        my $storage = prepare_storage($vars{storage_name}, old_random_timestamp());
-        push @old_versions, $storage;
-      }
-      my $storage = prepare_storage($vars{storage_name}, new_random_timestamp());
+    it "deletes nothing with a single old version and single new one" => sub {
+      my $old_storage = prepare_storage($vars{storage_name}, old_random_timestamp());
+      my $new_storage = prepare_storage($vars{storage_name}, new_random_timestamp());
       my $exp = HTFeed::BackupExpiration->new(storage_name => $vars{storage_name});
       $exp->run();
       my $deleted = count_deleted_objects('test', 'test');
-      is($deleted, 2, 'two old objects marked with feed_backups.deleted=1');
-      $deleted = count_deleted_objects('test', 'test', $storage->{timestamp});
-      is($deleted, 0, 'newest object still feed_backups.deleted=0');
-      foreach my $old_storage (@old_versions) {
-        ok(mets_deleted($old_storage), "old ($old_storage->{timestamp}) mets deleted");
-        ok(zip_deleted($old_storage), "old ($old_storage->{timestamp}) zip deleted");
-      }
-      ok(!mets_deleted($storage), "new ($storage->{timestamp}) mets left intact");
-      ok(!zip_deleted($storage), "new ($storage->{timestamp}) zip left intact");
+      is($deleted, 0, 'no objects deleted');
+      ok(!mets_deleted($old_storage), "old ($old_storage->{timestamp}) mets left intact");
+      ok(!zip_deleted($old_storage), "old ($old_storage->{timestamp}) zip left intact");
+      ok(!mets_deleted($new_storage), "new ($new_storage->{timestamp}) mets left intact");
+      ok(!zip_deleted($new_storage), "new ($new_storage->{timestamp}) zip left intact");
     };
 
-    it "keeps the newest old version" => sub {
+    it "deletes the oldest old versions when there are no new ones" => sub {
       my @old_versions;
-      foreach my $n (1 .. 2) {
+      foreach my $n (1 .. 3) {
         my $storage = prepare_storage($vars{storage_name}, old_random_timestamp());
         push @old_versions, $storage;
       }
-      my ($older, $newer) = @old_versions;
-      if ($old_versions[0]->{timestamp} > $old_versions[1]->{timestamp}) {
-        ($newer, $older) = @old_versions;
-      }
+      @old_versions = sort { $a->{timestamp} cmp $b->{timestamp}; } @old_versions;
```

Comment:
This looks correct; I wonder though if it would be clearer if we explicitly specified the timestamps for each object rather than including this logic here.